Client/Frontend/UserContent/UserScripts/PaymentRequest.js

```diff
@@ -0,0 +1,36 @@
+paymentreq_reject = {}
+paymentreq_resolve = {}
+
+Object.defineProperty(window, 'paymentreq_postCreate', {
```

Comment:
Is it necessary to define `paymentreq_postCreate` on window? OR can we do:

```javascript
window.paymentreq_postCreate = function(response) {
    paymentreq_resolve(response)
}
```

Reply:
@Brandon-T - `Object.defineProperty` sets the writable configuration of the property to `false`. 

With the above code, an attacker can re-define `window.paymentreq_postCreate` and may be able to modify the response from the client before sending the response back to the client.

Reply:
@jumde if this is true, and we are relying on this for protecting users, then we have a security flaw in this code because:

1. Create localhost server (used flask in python3).
2. In index.html (or any page) inside of the <head></head> tag, I added:
```javascript
<script>
Object.defineProperty(window, 'paymentreq_postCreate', {
            value:
                function (response) {
                    console.log("Hacked..");
                },
            writable: false,
            configurable: false
        })
</script>
```
3. On iOS, inject a script like yours `onDocumentStart: true, mainFrameOnly: false`:
```javascript
Object.defineProperty(window, 'paymentreq_postCreate', {
    value:
        function (response) {
            console.log("INJECTED Function from iOS");
        },
    writable: false,
    configurable: false
})

console.log("LOADED U2F script\n");
```

Now in safari debugger, I did: `window.paymentreq_postCreate(0);` and it prints `Hacked..` instead of `INJECTED Function from iOS`..

That's because iOS' `onDocumentStart` means that the script is injected AFTER the document element is created: https://developer.apple.com/documentation/webkit/wkuserscriptinjectiontime/atdocumentstart
This means, the page's javascript will get executed before our injected scripts (as long as the page's javascript is in the head tag).

Reply:
This is helpful, thanks @Brandon-T . Will update!